community/command-line/src/main/java/org/neo4j/commandline/arguments/Arguments.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    public Arguments withOptionalPositionalArgument( int position, String value )
+    {
+        return withPositionalArgument( new OptionalPositionalArgument( position, value ) );
+    }
+
+    public Arguments withMandatoryPositionalArgument( int position, String value )
+    {
+        return withPositionalArgument( new MandatoryPositionalArgument( position, value ) );
+    }
+
+    public Arguments withArgument( NamedArgument namedArgument )
+    {
+        namedArgs.put( namedArgument.name(), namedArgument );
+        return this;
+    }
+
+    public Arguments withPositionalArgument( PositionalArgument arg )
+    {
+        positionalArgs.add( arg );
+        return this;
+    }
+
+    public String usage()
+    {
+        StringBuilder sb = new StringBuilder();
+
+        if ( !namedArgs.isEmpty() )
+        {
+            sb.append( namedArgs.values().stream().map( NamedArgument::usage ).collect( Collectors.joining( " " ) ) );
+        }
+
+        if ( !positionalArgs.isEmpty() )
+        {
+            sb.append( " " );
+            positionalArgs.sort( ( l, r ) -> Integer.compare( l.position(), r.position() ) );
+            sb.append( positionalArgs.stream().map( a -> a.usage() ).collect( Collectors.joining( " " ) ) );
+        }
+
+        return sb.toString().trim();
+    }
+
+    public String description( String text )
+    {
+        String wrappedText = WordUtils.wrap( text, LINE_LENGTH );
+        if ( namedArgs.isEmpty() )
+        {
+            return wrappedText;
+        }
+
+        wrappedText = String.join( "\n\n", wrappedText, "options:" );
+
+        //noinspection OptionalGetWithoutIsPresent handled by if-statement above
+        final int alignLength = namedArgs.values().stream()
+                .map( a -> a.optionsListing().length() )
+                .reduce( 0, Integer::max );
+
+        return String.join( "\n", wrappedText,
+                namedArgs.values().stream()
+                        .map( c -> formatArgumentDescription( alignLength, c ) )
+                        .collect( Collectors.joining( "\n" ) ) );
+    }
+
+    public String formatArgumentDescription( final int longestAlignmentLength, final NamedArgument argument )
+    {
+        final String left = String.format( "  %s", argument.optionsListing() );
+        final String right;
+        if ( argument instanceof OptionalNamedArg )
+        {
+            right = String.format( "%s [default:%s]", argument.description(),
+                    ((OptionalNamedArg) argument).defaultValue() );
+        }
+        else
+        {
+            right = argument.description();
+        }
+        return rightColumnFormatted( left, right, longestAlignmentLength + 5 );
+    }
+
+    public static String rightColumnFormatted( final String leftText, final String rightText, int rightAlignIndex )
+    {
+        int rightWidth = Arguments.LINE_LENGTH - rightAlignIndex;
+        boolean startOnNewLine = false;
+        if ( rightWidth < 30 )
```

Comment:
I picked a number at random.

Then I thought about it.. and changed it.

It's the minimum size of the right column. If less, it prints the description on a new line instead (see import command for example)